datafusion/physical-expr/src/binary_map.rs

```diff
@@ -364,7 +364,9 @@ where
                     let null_index = self.offsets.len() - 1;
                     // nulls need a zero length in the offset buffer
                     let offset = self.buffer.len();
-                    self.offsets.push(O::from_usize(offset).unwrap());
+                    // safety: check for overflows at the end
+                    let offset = unsafe { O::from_usize_unchecked(offset)} ;
```

Comment:
this skips overflow checking so if the buffer length is larger than can fit in a i32 it will overflow on strings

I added a check at the end